src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -104,7 +105,7 @@ class GoogleDataprocDAO(protected val dataprocConfig: DataprocConfig,
     new GoogleCredential.Builder()
       .setTransport(httpTransport)
       .setJsonFactory(jsonFactory)
-      .setServiceAccountId(dataprocConfig.serviceAccountEmail.value)
+      .setServiceAccountId(serviceAccountProvider.getLeoServiceAccount.value)
```

Comment:
if this is the only place where we use `serviceAccountProvider` can we just pass in the leo SA in the ctor instead

Reply:
+1